app/controllers/sidebar_sections_controller.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+class SidebarSectionsController < ::ApplicationController
+  requires_login
+
+  def create
+    sidebar_section =
+      SidebarSection.create!(
+        section_params.merge(user: current_user, sidebar_urls_attributes: links_params),
+      )
+
+    render json: SidebarSectionSerializer.new(sidebar_section)
+  rescue ActiveRecord::RecordInvalid => e
+    render_json_error(e.record.errors.full_messages.first)
+  end
+
+  def update
+    sidebar_section = SidebarSection.find(section_params["id"])
+    @guardian.ensure_can_edit!(sidebar_section)
+
+    sidebar_section.update!(
+      section_params.merge(user: current_user, sidebar_urls_attributes: links_params),
```

Comment:
I don't think we need `user: current_user` here when we're updating since we've already validated that the section belongs to the current user.